### PR TITLE
Fuse index

### PR DESCRIPTION
**Pull request by thargosu**
Hi,

This is an initial approach to using a Fuse.js index and debouncing to improve the performance of the search feature. It is accompanied by another pull request for the org-roam-ui project: 
https://github.com/ikoamu/org-roam-ui/pull/14

I believe these changes have improved the performance of my project, and I would appreciate your feedback. I think this could help in resolving issue #24.

Thank you for your time!

### CODE REVIEW

File: local.sh [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
 
-read -p "Enter the ROAM_PATH: " ROAM_PATH
-read -p "Enter the ROAM_DB_FILE: " ROAM_DB_FILE
+# export ROAM_PATH=/path/to/org-roam/src
+# export ROAM_DB_FILE=org-roam.db
```
Comment (ikoamu) [inline local.sh:4]: Would it be alright if you restore the comment here?

**Review by ikoamu — COMMENTED** [review]
Please add fuse-index.json to .gitignore to exclude it from Git tracking.

**Comment by ikoamu** [comment]
Thank you so much for submitting such an excellent Pull Request! I'm very grateful.
I've confirmed that there are no functional issues, but I've made a few minor comments. Please address them.

Also, I apologize for not being able to attend to the issue sooner.
I didn't have time due to work on other projects.

In any case, I'm very happy if this PR improves the search performance. Thank you!!!!!

File: local.sh (hunk 1 above)
Comment (thargosu) [reply local.sh:4]: Sure! I will restore the previous state here :+1:

**Comment by thargosu** [comment]
Thanks for your feedback!

No worries about not having the time to attend the issue. There is no rush. I had the opportunity to dive into it so I gave it a try.
I'm happy that you do not see any functional issues.

I will address your comments as soon as possible.

**Comment by thargosu** [comment]
I have added fuse-index.json to the .gitignore and restore the state of the local.sh script.
Feel free to get back to me if you see additional corrections to be made.
I'm happy I can participate a little in this really great project.

**Review by ikoamu — COMMENTED** [review]
LGTM!
thank you very much.

**Review by ikoamu — APPROVED** [review]

**Pull request merged**